### ISSUE TITLE
Warning and Build Failure during 'yarn visionos' Execution

### ISSUE THREAD

**Issue by sstefdev** [callstack/react-native-visionos#110]
### Description

When executing the command yarn visionos in the project directory, a warning and subsequent build failure occur. The warning pertains to the "Run script build phase 'Bundle React Native code and images'", indicating potential inefficiency or redundancy in the build process. Furthermore, the build fails with an error related to the compilation of 'RCTXRModule.o' during the 'React-RCTXR' target build process.

### OS
macOS Sonoma Version 14.2 (23C64)

### Device
Apple M1 Pro 2021

### Steps to reproduce

All initial steps were finished without errors.

- Navigate to the project directory.
- Execute the command: yarn visionos.

### React Native Version

0.73.4

### Affected Platforms

Runtime - iOS

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.2
  CPU: (10) arm64 Apple M1 Pro
  Memory: 105.70 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.3.0
    path: /opt/homebrew/bin/node
  Yarn:
    version: 3.6.4
    path: /opt/homebrew/bin/yarn
  npm:
    version: 9.6.7
    path: /opt/homebrew/bin/npm
  Watchman:
    version: 2024.01.22.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.2
      - iOS 17.2
      - macOS 14.2
      - tvOS 17.2
      - visionOS 1.0
      - watchOS 10.2
  Android SDK: Not Found
IDEs:
  Android Studio: Not Found
  Xcode:
    version: 15.2/15C500b
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 11.0.20
    path: /usr/bin/javac
  Ruby:
    version: 3.3.0
    path: /Users/stefdev/.asdf/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react: Not Found
  react-native: Not Found
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: false
```

### Stacktrace or Logs

```text
The following build commands failed:
        CompileC /Users/stefdev/Library/Developer/Xcode/DerivedData/VisioWallet-gcvxkpbjwjyvtpfrtntbbdqbxbva/Build/Intermediates.noindex/Pods.build/Debug-xrsimulator/React-RCTXR.build/Objects-normal/arm64/RCTXRModule.o /Users/stefdev/Desktop/VisioWallet/node_modules/@callstack/react-native-visionos/Libraries/XR/RCTXRModule.mm normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'React-RCTXR' from project 'Pods')
(1 failure)
```

### Reproducer

N/A

**Comment by okwasniewski** [comment]
Hey!

I've released new version 0.73.4-1 with a fix for another issue that was causing build issues (#106). The provided error message is not descriptive enough, unfortunately. Please try on the newer version (first clean build cache and Xcode derived data) and if that won't help please create a reproduction repository.

**Comment by sstefdev** [comment]
Yeah the new version works fine now, thank you
